### PR TITLE
Put everything into an Envoy namespace

### PR DESCRIPTION
**Pull request by tschroed**
- Envoy namespace all around
- Update check_format
- Update documentation
- Advance build_setup envoy-filter-example hash to the corresponding change

### CODE REVIEW

File: tools/check_format.py [hunk 1]
```diff
@@ -24,6 +25,16 @@ def printError(error):
...
 
 
+def checkNamespace(file_path):
+  with open(file_path) as f:
+    text = f.read()
+    if not re.search('^\s*namespace\s+Envoy\s*{', text, re.MULTILINE) and \
+       not re.search('NOLINT\(namespace-envoy\)', text, re.MULTILINE):
+      printError("Unable to find Envoy namespace or NOLINT(namespace-envoy) for file: %s" % file_path)
+      return False
+  return True
```
Comment (htuch) [inline tools/check_format.py]: Return value unused.

File: tools/check_format.py [hunk 2]
```diff
@@ -24,6 +25,16 @@ def printError(error):
   print "ERROR: %s" % (error)
 
 
+def checkNamespace(file_path):
+  with open(file_path) as f:
+    text = f.read()
+    if not re.search('^\s*namespace\s+Envoy\s*{', text, re.MULTILINE) and \
```
Comment (htuch) [inline tools/check_format.py]: Why so relaxed on the number of whitespaces?

File: STYLE.md [hunk 3]
```diff
@@ -35,6 +35,10 @@
   developer discretion. Where possible, methods should have meaningful
   documentation on expected input and state preconditions.
 * Header guards should use `#pragma once`.
+* All code should be inside a top-level Envoy namespace. There are some
+  exceptions such as main() functions. When code cannot be placed inside the
```
Comment (htuch) [inline STYLE.md]: Nit: `main()` (Markdown inline code)

**Comment by htuch** [comment]
git tip: squash down all commits before first review, it's a bit less noisy.

File: tools/check_format.py (hunk 1 above)
Comment (tschroed) [inline tools/check_format.py]: It's used in fixFilePath.

File: tools/check_format.py (hunk 2 above)
Comment (tschroed) [inline tools/check_format.py]: Clang-tidy will figure out the right number of spaces. I just wanted to ensure that the line was present at all.

**Pull request merged**